### PR TITLE
MYS 192 runtime type checking

### PR DESCRIPTION
**Pull request by mipo47**
Some tests still contains runtime type errors, that's why checking is turned off now.

To turn type checking open .babelrc and change: 
"assert": false to "assert": true

Check out this branch and run 'yarn' to ensure, that your node-js version is supported

### CODE REVIEW

File: .babelrc [hunk 1]
```diff
@@ -12,7 +12,14 @@
         }],
         "stage-0"
       ],
-      "plugins": ["istanbul"]
+      "plugins": [
+        "istanbul",
+        "transform-decorators-legacy",
```
Comment (donce) [inline .babelrc]: ?? Is this needed by flow-runtime?

File: src/app/statistics/elk-collector.js [hunk 2]
```diff
@@ -20,7 +20,7 @@ import axios from 'axios'
 import type { Event, EventCollector } from './events'
 
 class ElkCollector implements EventCollector {
-  _axiosApi: axios.Axios
+  _axiosApi: any
```
Comment (donce) [inline src/app/statistics/elk-collector.js]: Maybe we can create a local type definition for methods we use?

File: src/libraries/mysterium-tequilapi/dto/identity-registration.js [hunk 3]
```diff
@@ -16,6 +16,7 @@
  */
 
 // @flow
+// @flow-runtime
```
Comment (donce) [inline src/libraries/mysterium-tequilapi/dto/identity-registration.js]: ?? Do these work with flow-runtime disabled?

File: src/renderer/store/modules/connection.js [hunk 4]
```diff
@@ -37,11 +37,11 @@ import type { Provider } from '../../../app/connection/provider'
 import messages from '../../../app/messages'
 
 type ConnectionStore = {
-  ip: ?string,
-  location: ?ConsumerLocationDTO,
+  ip?: ?string,
```
Comment (donce) [inline src/renderer/store/modules/connection.js]: ?? Why 2 question marks? One is usually enough.

Comment (mipo47) [reply src/renderer/store/modules/connection.js]: one question mark to allow skip property,
another one to make property nullable
(no way to make it with one question mark)

File: src/libraries/mysterium-tequilapi/dto/identity-registration.js (hunk 3 above)
Comment (mipo47) [reply src/libraries/mysterium-tequilapi/dto/identity-registration.js]: hopefully no, I'll remove it

File: src/app/statistics/elk-collector.js (hunk 2 above)
Comment (mipo47) [reply src/app/statistics/elk-collector.js]: sounds like great future task

File: .babelrc (hunk 1 above)
Comment (mipo47) [reply .babelrc]: Can't remember why I've added it, but now it's not necessary

File: src/renderer/store/modules/connection.js (hunk 4 above)
Comment (donce) [reply src/renderer/store/modules/connection.js]: Maybe we can have `ip: ?string` and initialise this with `{ ip: null }` ?

File: src/app/communication/dto.js [hunk 5]
```diff
@@ -22,7 +22,7 @@ import type { Country } from '../countries'
 import type { Metric } from '../bug-reporting/metrics/metrics'
 
 export type ConnectionStatusChangeDTO = {
-  oldStatus: ConnectionStatus,
+  oldStatus: ?ConnectionStatus,
   newStatus: ConnectionStatus
```
[the remark this answers is not in the document]
Comment (mipo47) [reply src/app/communication/dto.js]: before first request you have no idea about status, that's why it's nullable.
New status is always latest status, can't be null

File: src/renderer/store/modules/connection.js (hunk 4 above)
Comment (mipo47) [reply src/renderer/store/modules/connection.js]: don't want to change current code, so I just fixed type definition

**Pull request merged**